### PR TITLE
Add remove_source_branch_after_merge in project api

### LINKED ISSUE #742: Implement remove_source_branch_after_merge
Please implement the new ([12.5](https://about.gitlab.com/blog/2019/11/22/gitlab-12-5-released/#delete-source-branch-once-merged)) feature `remove_source_branch_after_merge`.

### PR DESCRIPTION
**Pull request by flolivaud**
Close #742

### CODE REVIEW

File: projects.go [hunk 1]
```diff
@@ -472,6 +472,7 @@ type CreateProjectOptions struct {
 	UseCustomTemplate                         *bool             `url:"use_custom_template,omitempty" json:"use_custom_template,omitempty"`
 	GroupWithProjectTemplatesID               *int              `url:"group_with_project_templates_id,omitempty" json:"group_with_project_templates_id,omitempty"`
 	BuildCoverageRegex                        *string           `url:"build_coverage_regex,omitempty" json:"build_coverage_regex,omitempty"`
+	RemoveSourceBranchAfterMerge              *bool             `url:"remove_source_branch_after_merge,omitempty" json:"remove_source_branch_after_merge,omitempty"`
```
Comment (svanharmelen) [inline projects.go]: Could you reorder these a little to keep the ordering inline with the docs? That would mean moving the `BuildCoverageRegex` field above the `CIConfigPath` and the new `RemoveSourceBranchAfterMerge` field that you just added above `LFSEnabled`. Would be great if we can do that before we merge it, thanks!

File: projects.go [hunk 2]
```diff
@@ -556,6 +557,7 @@ type EditProjectOptions struct {
 	MirrorOverwritesDivergedBranches          *bool             `url:"mirror_overwrites_diverged_branches,omitempty" json:"mirror_overwrites_diverged_branches,omitempty"`
 	PackagesEnabled                           *bool             `url:"packages_enabled,omitempty" json:"packages_enabled,omitempty"`
 	BuildCoverageRegex                        *string           `url:"build_coverage_regex,omitempty" json:"build_coverage_regex,omitempty"`
+	RemoveSourceBranchAfterMerge              *bool             `url:"remove_source_branch_after_merge,omitempty" json:"remove_source_branch_after_merge,omitempty"`
```
Comment (svanharmelen) [inline projects.go]: Same about the ordering...

**Comment by Andor** [comment]
#742

File: projects.go (hunk 1 above)
Comment (flolivaud) [reply projects.go]: I will do it next week.

File: projects.go (hunk 2 above)
Comment (flolivaud) [reply projects.go]: Same as above

**Comment by flolivaud** [comment]
I had reordered the options for "create" and "edit" project api.

**Pull request merged**